tests/setup_test.go

```diff
@@ -1104,3 +1104,184 @@ func TestDirectRunVariables(t *testing.T) {
+			delete: true,
+		},
+		{
+			name:   "test delete log with unexisting step",
+			step:   99,
+			delete: true,
+			err:    true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "agola")
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			defer os.RemoveAll(dir)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			tetcd, tgitea, c := setup(ctx, t, dir)
+			defer shutdownGitea(tgitea)
+			defer shutdownEtcd(tetcd)
+
+			gwClient := gwclient.NewClient(c.Gateway.APIExposedURL, "admintoken")
+			user, _, err := gwClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: agolaUser01})
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			t.Logf("created agola user: %s", user.UserName)
+
+			token := createAgolaUserToken(ctx, t, c)
+
+			// From now use the user token
+			gwClient = gwclient.NewClient(c.Gateway.APIExposedURL, token)
+
+			directRun(t, dir, config, c.Gateway.APIExposedURL, token)
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				runs, _, err := gwClient.GetRuns(ctx, nil, nil, []string{path.Join("/user", user.ID)}, nil, "", 0, false)
+				if err != nil {
+					return false, nil
+				}
+
+				if len(runs) != 1 {
+					return false, nil
+				}
+
+				run := runs[0]
+				if run.Phase != rstypes.RunPhaseFinished {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			runs, _, err := gwClient.GetRuns(ctx, nil, nil, []string{path.Join("/user", user.ID)}, nil, "", 0, false)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			t.Logf("runs: %s", util.Dump(runs))
+
+			if len(runs) != 1 {
+				t.Fatalf("expected 1 run got: %d", len(runs))
+			}
+
+			run, _, err := gwClient.GetRun(ctx, runs[0].ID)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			if run.Phase != rstypes.RunPhaseFinished {
+				t.Fatalf("expected run phase %q, got %q", rstypes.RunPhaseFinished, run.Phase)
+			}
+			if run.Result != rstypes.RunResultSuccess {
+				t.Fatalf("expected run result %q, got %q", rstypes.RunResultSuccess, run.Result)
+			}
+
+			var task *gwapitypes.RunResponseTask
+			for _, t := range run.Tasks {
+				if t.Name == "task01" {
+					task = t
+					break
+				}
+			}
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				t, _, err := gwClient.GetRunTask(ctx, runs[0].ID, task.ID)
+				if err != nil {
+					return false, nil
+				}
+				if !t.Steps[tt.step].LogArchived {
+					return false, nil
+				}
+				return true, nil
+			})
+
+			if tt.delete {
+				_, err := gwClient.DeleteLogs(ctx, run.ID, task.ID, tt.setup, tt.step)
+				if err != nil {
+					if !tt.err {
+						t.Fatalf("log delete error: %v", err)
+					}
+				}
+			}
+
+			_, err = gwClient.GetLogs(ctx, run.ID, task.ID, tt.setup, tt.step)
+			if tt.delete {
+				if err == nil {
+					if tt.err {
+						t.Fatalf("unexpected err: %v", err)
+					}
+				}
```

Comment:
This will print `unexpected err: nil` and won't cover the other cases. You should use something like this check:

https://github.com/agola-io/agola/blob/master/internal/config/config_test.go#L206-L223

 and set err as type error so you'll also check the value